test/test_jit.py

```diff
@@ -2105,6 +2105,19 @@ def reassign():
             return
         self.checkScript(reassign, (), optimize=True)
 
+        def shape_analysis(x):
+            y = [x]
+            if True:
+                y = [x, x]
+            while False:
+                y = [x, x, x]
+            return torch.cat(y, dim=1)
+
+        x = torch.zeros(2, 4, 5)
+        graph = torch.jit.script(shape_analysis).graph
+        torch._C._jit_pass_shape_analysis(graph, (x,), False)
+        self.assertExpected(str(graph))
```

Comment:
Use `assertExpectedGraph` and `run_pass`. I also don't really know if this deserves a test, and it should definitely not be part of the `test_list_literal`, because that's not it's purpose!

Reply:
run_pass doesn't work with shape_analysis because there are tensor inputs